nodebuilder/tests/share_test.go

```diff
@@ -0,0 +1,279 @@
+//go:build share || integration
+
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	libshare "github.com/celestiaorg/go-square/v2/share"
+
+	"github.com/celestiaorg/celestia-node/blob"
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/nodebuilder/tests/swamp"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+func TestShareModule(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 25*time.Second)
+	t.Cleanup(cancel)
+	sw := swamp.NewSwamp(t, swamp.WithBlockTime(time.Second*1))
+	blobSize := 128
+	libBlob, err := libshare.GenerateV0Blobs([]int{blobSize}, true)
+	require.NoError(t, err)
+
+	nodeBlob, err := blob.ToNodeBlobs(libBlob[0])
+	require.NoError(t, err)
+
+	bridge := sw.NewBridgeNode()
+	require.NoError(t, bridge.Start(ctx))
+	addrs, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(bridge.Host))
+	require.NoError(t, err)
+
+	fullCfg := sw.DefaultTestConfig(node.Full)
+	fullCfg.Header.TrustedPeers = append(fullCfg.Header.TrustedPeers, addrs[0].String())
+	fullNode := sw.NewNodeWithConfig(node.Full, fullCfg)
+	require.NoError(t, fullNode.Start(ctx))
+
+	addrsFull, err := peer.AddrInfoToP2pAddrs(host.InfoFromHost(fullNode.Host))
+	require.NoError(t, err)
+
+	lightCfg := sw.DefaultTestConfig(node.Light)
+	lightCfg.Header.TrustedPeers = append(lightCfg.Header.TrustedPeers, addrsFull[0].String())
+	lightNode := sw.NewNodeWithConfig(node.Light, lightCfg)
+	require.NoError(t, lightNode.Start(ctx))
+
+	bridgeClient := getAdminClient(ctx, bridge, t)
+	fullClient := getAdminClient(ctx, fullNode, t)
+	lightClient := getAdminClient(ctx, lightNode, t)
+
+	height, err := fullClient.Blob.Submit(ctx, nodeBlob, state.NewTxConfig())
+	require.NoError(t, err)
+
+	_, err = fullClient.Header.WaitForHeight(ctx, height)
+	require.NoError(t, err)
+	_, err = lightClient.Header.WaitForHeight(ctx, height)
+	require.NoError(t, err)
+
+	sampledBlob, err := fullClient.Blob.Get(ctx, height, nodeBlob[0].Namespace(), nodeBlob[0].Commitment)
+	require.NoError(t, err)
+
+	hdr, err := fullClient.Header.GetByHeight(ctx, height)
+	require.NoError(t, err)
+
+	coords, err := shwap.SampleCoordsFrom1DIndex(sampledBlob.Index(), len(hdr.DAH.RowRoots))
+	require.NoError(t, err)
+
+	blobAsShares, err := blob.BlobsToShares(sampledBlob)
+	require.NoError(t, err)
+
+	test := []struct {
+		name string
+		doFn func(t *testing.T)
+	}{
+		{
+			name: "SharesAvailable",
+			doFn: func(t *testing.T) {
+				err := lightClient.Share.SharesAvailable(ctx, height)
+				require.NoError(t, err)
+				err = fullClient.Share.SharesAvailable(ctx, height)
+				require.NoError(t, err)
+				err = bridgeClient.Share.SharesAvailable(ctx, height)
+				require.NoError(t, err)
```

Comment:
Why not loop over every client? Very table-driven test coded

Reply:
yeap. will look much better 👍